### PR TITLE
Support reference datetime for examining historical dhcpd.leases

### PR DESCRIPTION
**Pull request by aaron-sierra**
While debugging a DHCP issue, I found myself needing to review historical `dhcpd.leases` files. It took me a while to realize it, but as time passed the results from my review started to change :frowning_face: 

I found that this module was checking lease validity against the current system time and that testing used `@freeze_time` decorators.

In order to do my work, I modified the module to also support a user-specified reference `datetime` object when initializing classes. I think that it could be useful to others, too.

### CODE REVIEW

**Comment by aaron-sierra** [comment]
In reviewing my work, I noticed that the unit tests that I'd picked for the `Lease` and `Lease6` classes weren't sensitive to changes in the "now" time, so I moved testing from `test_init()` to `test_valid()`.

File: isc_dhcp_leases/iscdhcpleases.py [hunk 1]
```diff
@@ -108,9 +111,13 @@ class IscDhcpLeases(object):
         r"ia-(?P<type>ta|na|pd) \"(?P<id>[^\"\\]*(?:\\.[^\"\\]*)*)\" {(?P<config>[\s\S]+?)\n}")
     regex_iaaddr = re.compile(r"ia(addr|prefix) (?P<ip>[0-9a-f:]+(/[0-9]+)?) {(?P<config>[\s\S]+?)\n\s+}")
 
-    def __init__(self, filename, gzip=False):
+    def __init__(self, filename, gzip=False, now=None):
+        if not datetime_is_ok(now):
+            raise ValueError
```
Comment (andir) [inline isc_dhcp_leases/iscdhcpleases.py]: Can't you pass a descriptive error message to ValueError? I think the user shouldn't have to dive through the stack trace if things go wrong.

Comment (aaron-sierra) [reply isc_dhcp_leases/iscdhcpleases.py]: Yeah, that's a good point. Thanks for the review, I'll push an update shortly.

Comment (aaron-sierra) [reply isc_dhcp_leases/iscdhcpleases.py]: I converted `datetime_is_ok()` (returning boolean) to `check_datetime()` (directly raising `ValueError` when appropriate).

I also tweaked the test to *really* only accept `None` or timezone-aware datetime objects.

**Comment by MartijnBraam** [comment]
Looks good, thanks for the addition
